lib/PrivateAppUserTokenManager.ts

```diff
@@ -0,0 +1,243 @@
+import {
+  fetchPrivateAppUserToken,
+  createPrivateAppUserToken,
+  updatePrivateAppUserToken,
+  PrivateAppUserTokenResponse,
+} from '../api/privateAppUserToken';
+import { scopeGroupsForPersonalAccessKey } from './personalAccessKey';
+import { logger } from './logger';
+import { i18n } from '../utils/lang';
+import moment from 'moment';
+import { AxiosError, isAxiosError } from 'axios';
+import { getAxiosErrorWithContext } from '../errors/apiErrors';
+import { throwErrorWithMessage } from '../errors/standardErrors';
+import { BaseError } from '../types/Error';
+
+const USER_TOKEN_READ = 'developer.private_app.temporary_token.read';
+const USER_TOKEN_WRITE = 'developer.private_app.temporary_token.write';
+const i18nKey = 'lib.PrivateAppUserTokenManager';
+
+export class PrivateAppUserTokenManager {
+  accountId: number;
+  private tokenMap: Map<number, PrivateAppUserTokenResponse>;
+  private tokenMapIntervalId: Map<number, NodeJS.Timeout>;
+  private enabled: boolean;
+
+  constructor(accountId: number) {
+    this.accountId = accountId;
+    this.tokenMap = new Map<number, PrivateAppUserTokenResponse>();
+    this.tokenMapIntervalId = new Map<number, NodeJS.Timeout>();
+    this.enabled = false;
+  }
+
+  async init(): Promise<void> {
+    const scopeGroups = new Set<string>(
+      await scopeGroupsForPersonalAccessKey(this.accountId)
+    );
+    if (
+      !scopeGroups.has(USER_TOKEN_READ) ||
+      !scopeGroups.has(USER_TOKEN_WRITE)
+    ) {
+      throwErrorWithMessage(`${i18nKey}.errors.noScopes`, {
+        accountId: this.accountId,
+      });
+    } else {
+      logger.debug(
+        i18n(`${i18nKey}.enabled`, {
+          accountId: this.accountId,
+        })
+      );
+      this.enabled = true;
+    }
+  }
+
+  cleanup() {
+    this.tokenMapIntervalId.forEach(timeoutId => clearInterval(timeoutId));
+    this.tokenMapIntervalId.clear();
+    this.tokenMap.clear();
+  }
+
+  async getPrivateAppToken(
+    appId: number,
+    scopeGroups: string[] = []
+  ): Promise<string | undefined> {
+    if (!this.enabled) {
```

Comment:
Should this throw? I'm assuming we wouldn't want the user to be able to continue if we hit this case for some reason

cc/ @camden11

Reply:
Yeah, I guess it ultimately depends on how this returning `undefined` is handled in the CLI, but might make more sense to turn this debug into an error

Reply:
If it's not enabled, it would function like it does now. The localdev server could work if the PAT is within their env file.